src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +35,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        //this.value = address;
+        String[] addressArr = address.split(",");
+        this.block = new Block(addressArr[ADDRESS_BLOCK_INDEX]);
+        this.street = new Street(addressArr[ADDRESS_STREET_INDEX]);
```

Comment:
note that address is separate by ", ". So when you do this, your street will have an extra space. How do you handle that? :)